### DISCUSSION TITLE
Prefil select

### DISCUSSION THREAD

**Discussion by gregoryloichot** [filamentphp/filament#13383] [Help] [answered]
### Package

Form builder

### Package Version

v3.2

### How can we help you?

Hello, 

I cannot find the best solution to prefill a SelectBox (I'm still exploring FilamentPHP; which turns out to be great :))

In a relation manager (CommentRelationManager), I have:

```php
public function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\TextInput::make('name')
                    ->required()
                    ->maxLength(255),
                Forms\Components\Select::make('article_id')
                    ->required()
                    ->native(false)
                    ->preload()
                    ->relationship('article', 'name'),
…
```

What I wanna do: to make `article_id` equals to parent record's id (model `Article` is the parent of the current `Comment`). I tried the `afterStateHydrated` method but I did not success, and I am not sure it is the correct solution.

If anyone could help me 🙏

**Comment by gregoryloichot** [comment]
Well… after hours, I found `formatStateUsing` :

```php
public function form(Form $form): Form
    {
        return $form
            ->schema([
                Forms\Components\TextInput::make('name')
                    ->required()
                    ->maxLength(255),
                Forms\Components\Select::make('article_id')
                    ->required()
                    ->native(false)
                    ->preload()
                    ->relationship('article', 'name'),
                    ->formatStateUsing(fn(?string $state): int => $this->getOwnerRecord()->id),
```

Might someone confirm it is the best solution?

**Reply by bubu-bin** [reply]
Yes it's the only one that worked for me.

**Comment by dissto** [comment, marked as answer]
Mhh, unless you want the article_id to be chosen by the user, you could [mutate the data](https://filamentphp.com/docs/3.x/actions/prebuilt-actions/create#customizing-data-before-saving) instead.

Perhaps something like that on your `CommentRelationManager`'s create action

```php
Tables\Actions\CreateAction::make()
  ->mutateFormDataUsing(function (array $data, $livewire) {
      $data['article_id'] = $livewire->getOwnerRecord()->getKey();
      return $data;
  })
```
Then you could completely remove the select from the form. Or maybe I misunderstood what you are trying to achieve 🤔 

Or do you actually just want to set a default value? 
```php
Forms\Components\Select::make('article_id')
  ->default($this->getOwnerRecord()->getKey())
```

**Reply by gregoryloichot** [reply]
Yes, you are right.

I think I kept the select to be consistent with an other view that allow the user to modify the value of the Article. 

But yes, in this case, I should probably remove the selectbox.
